assets/src/components/parts/janus-capi-iframe/ExternalActivity.tsx

```diff
@@ -520,7 +529,23 @@ const ExternalActivity: React.FC<PartComponentProps<CapiIframeModel>> = (props)
     const filterVars = createCapiObjectFromStateVars(simLife.currentState, simLife.domain);
     if (filterVars && Object.keys(filterVars)?.length !== 0) {
       handleIFrameSpecificProperties(simLife.currentState, simLife.domain);
-      sendFormedResponse(simLife.handshake, {}, JanusCAPIRequestTypes.VALUE_CHANGE, filterVars);
+      writeCapiLog('SENDING SIM CONFIG DATA !!!!', 3, {
+        simLife,
+        configData: filterVars,
+      });
+      Object.keys(filterVars).forEach((variable: any) => {
+        const formatted: Record<string, unknown> = {};
+        const simVariable = filterVars[variable];
+        const typeOfValue = typeof simVariable.value;
+        //Now we know that configData variable can contain a expression in value field so we need to parse that before sending it to SIM
+        if (typeOfValue === 'string' && simVariable.value.indexOf('{') !== -1) {
+          const gotid = templatizeText(simVariable.value, simLife.snapshot, scriptEnv);
+          simVariable.value = gotid;
+        }
+
+        formatted[variable] = simVariable;
+        sendFormedResponse(simLife.handshake, {}, JanusCAPIRequestTypes.VALUE_CHANGE, formatted);
```

Comment:
so now we not only need to process the possible expressions in string values but also send them one at a time?

Reply:
Yes. During KIP investigation, we found that we need to send it one variable at a time and we were already sending the 'init state' variables like this. This one is configData. We never changed it to send one at  a time so refactored that code.